clients/baseclient/client_util.go

```diff
@@ -23,7 +23,11 @@ func shouldRetry(err error) bool {
 	if err == nil {
 		return false
 	}
-	isMatching, _ := regexp.MatchString(" EOF$", err.Error())
+	isMatching := strings.Contains(err.Error(), "retry is needed")
+	if isMatching {
+		return true
+	}
+	isMatching, _ = regexp.MatchString(" EOF$", err.Error())
```

Comment:
Error ignore?

Reply:
It was ignored so far. This error can occur only if the regex is invalid.